src/mds/OpenFileTable.cc

```diff
@@ -342,8 +342,15 @@ void OpenFileTable::commit(MDSInternalContextBase *c, uint64_t log_seq, int op_p
     mds->objecter->mutate(oid, oloc, op, snapc, ceph::real_clock::now(), 0,
 			  gather.new_sub());
 
+#if !defined(__FreeBSD__)
     ctl.journaled_update.merge(ctl.to_update);
     ctl.journaled_remove.merge(ctl.to_remove);
+#else
+    // XXX FreeBSD does not seem to support std::map::merge for C++17
+    // So work around it until it does.
+    ctl.journaled_update.insert(ctl.to_update.begin(), ctl.to_update.end());
```

Comment:
@wjwithagen could you use a clang version with C++17 support? it's not a limit of FreeBSD but a limit of compiler you are using, IIUC.

Reply:
@tchaikov 
I'm using --std=c++17 in the options... 
```
/usr/bin/c++  -DCEPH_LIBDIR=\"/usr/local/lib\" -DCEPH_PKGLIBDIR=\"/usr/local/lib/ceph\" -D_FILE_OFFSET_BITS=64 -I/home/jenkins/workspace/ceph-master/build/src/include -I/home/jenkins/workspace/ceph-master/src -isystem /usr/local/include -isystem /home/jenkins/workspace/ceph-master/build/include -I/usr/local/include/nss/nss -I/usr/local/include/nspr -isystem /home/jenkins/workspace/ceph-master/src/xxHash -isystem /home/jenkins/workspace/ceph-master/src/rapidjson/include -I/home/jenkins/workspace/ceph-master/src/dmclock/src -I/home/jenkins/workspace/ceph-master/src/dmclock/support/src -isystem /home/jenkins/workspace/ceph-master/src/googletest/googletest/include -I/home/jenkins/workspace/ceph-master/src/lua/src -I/home/jenkins/workspace/ceph-master/build/src/lua  -Wall -Wtype-limits -Wignored-qualifiers -Winit-self -Wpointer-arith -Werror=format-security -fno-strict-aliasing -fsigned-char -Wno-unknown-pragmas -Wno-unused-function -Wno-unused-local-typedef -Wno-varargs -Wno-gnu-designator -Wno-missing-braces -Wno-parentheses -Wno-deprecated-register  -ftemplate-depth-1024 -Wno-invalid-offsetof -Wnon-virtual-dtor -Wno-unknown-pragmas -Wno-inconsistent-missing-override -Wno-mismatched-tags -Wno-unused-private-field -Wno-address-of-packed-member -fdiagnostics-color=auto  -O0 -g -DBEAST_NO_BOOST_STRING_VIEW -fuse-ld=/usr/bin/ld.lld -Wno-unused-command-line-argument -fPIC   -DHAVE_CONFIG_H -D__CEPH__ -D_REENTRANT -D_THREAD_SAFE -D__STDC_FORMAT_MACROS -std=c++17 -o CMakeFiles/mds.dir/OpenFileTable.cc.o -c /home/jenkins/workspace/ceph-master/src/mds/OpenFileTable.cc
```
Sorry for the long line...

And Clang is 6.0.0. so that is the latest released fro FreeBSD: 
```
FreeBSD clang version 6.0.0 (branches/release_60 324090) (based on LLVM 6.0.0)
Target: x86_64-unknown-freebsd12.0
Thread model: posix
InstalledDir: /usr/bin
```
So I've posted a question on the toolchain list to ask what is failing here.

Also tried to include just about all includes that there are.
Even that did not fix any.

Reply:
you are right. libc++ still does not support "Splicing Maps and Sets", see https://libcxx.llvm.org/cxx1z_status.html , search for "p0083r3" .

Reply:
@tchaikov 
Thanx, added that to the commit message.